Fixed refinements

johnc303 opened the pull request:
Fixed issue with refinements not appearing. Fixed issue with duplicate sets of refinements appearing.

johnc303 committed:
Added low-tech solution to duplicate refinements

johnc303 committed:
Merge branch 'foundry-v10' into foundry-v10-jc

biz812 commented:
Looks good to me, one suggestion that occurs to me is that would there be value in have a common root key for this sort of "tracked state" so that every time we do this we have a place to put and we know that all the sub keys have a common purpose...
So instead of `this.system.statCreationRequestFor[statType]`
We would have something like
`this.system.stateTracking.statCreationRequestFor[statType]`
then we could continue to place new stuff and move existing stuff into there as we go, e.g. `this.system.stateTracking.someOtherThing[fooBar]`

Not necessary for this PR but something to consider perhaps?

johnc303 committed:
Added a new sessionState object to the actor template with comment. Refactored to use said state when creating a new actor & cloning in the stats.

johnc303 commented:
> Looks good to me, one suggestion that occurs to me is would there be value in having a common root key for this sort of "tracked state"? So that every time we do this we have a place to put and we know that all the sub keys have a common purpose... So instead of `this.system.statCreationRequestFor[statType]` We would have something like `this.system.stateTracking.statCreationRequestFor[statType]` then we could continue to place new stuff and move existing stuff into there as we go, e.g. `this.system.stateTracking.someOtherThing[fooBar]`
> 
> Not necessary for this PR but something to consider perhaps?

Good shout, I've added a new `sessionState` object and updated the code :-)
